fda/registration.py

```diff
@@ -0,0 +1,274 @@
+    if shifts_array:
+        return delta
+
+    # Computes the values with the final shift to construct the FDataBasis
+    return fd.shift(delta, ext=ext, tfine=tfine, **kwargs)
+
+
+def landmark_shift(fd, landmarks, location='minimize', ext='default',
+                   tfine=[], shifts_array=False, **kwargs):
+    r"""Perform a shift registration of the curves to align the landmarks at
+        the same mark time.
+
+        Args:
+            fd (:obj:`FDataBasis` or :obj:`FDataGrid`): Functional data object.
+            landmarks (array_like): List with the landmarks of the samples.
+            location (str or numeric): Defines the time where the landmarks will
+                be alligned. The possible values are:
+
+                * location='minimize': Minimizes the max shift, aligning the
+                  landmarks at :math:`\frac{1}{2}(max(\text{landmarks})+
+                  min(\text{landmarks}))`.
+                * location='mean': Aligns the landmarks at the mean.
+                * location='median': Aligns the landmarks at the median.
+                * location='middle': Aligns the landmarks at the middle of
+                  the domain range.
+                * If a number is passed it is used to align the landmarks.
+                The default value is minimize.
+            ext (str or int, optional): Controls the extrapolation mode for elements
+                not in the domain range.
+
+                * If ext=0 or 'default' uses the default method defined in the fd
+                  object.
+                * If ext=1 or 'extrapolation' uses the extrapolated values.
+                * If ext=2 or 'periodic' extends the domain range periodically.
+                * If ext=3 or 'const' uses the boundary value.
+                * If ext=4 or 'slice' avoids extrapolation restricting the
+                  domain.
+                The default value is 'default'.
+            tfine (array_like, optional): Set of points where the
+                functions are evaluated to obtain the discrete
+                representation of the object to integrate. If an empty list is
+                passed it calls numpy.linspace with bounds equal to the ones defined
+                in fd.domain_range and the number of points the maximum
+                between 201 and 10 times the number of basis plus 1.
+            shifts_array (bool, optional): If True returns an array with the
+                shifts instead of a :obj:`FDataBasis` with the registered
+                curves. Default sets to False.
+            **kwargs: Keyword arguments to be passed to :meth:`from_data`.
+    """
+
+    if len(landmarks) != fd.nsamples:
+        raise ValueError("landmark list ({}) must have the same length "
+                         "than the number of samples ({})"
+                         .format(len(landmarks), fd.nsamples))
+
+    landmarks = numpy.asarray(landmarks)
+
+    # Parses location
+    if location == 'minimize':
```

Comment:
Another enumerated type would work better here.

Reply:
As I told you, I'm thinking of changing this argument and accepting a numeric value, a function to be called with landmark's vector or, by default, take 0.5(min(landmarks) + max(landmarks)) that minimizes the max shift.

Reply:
Changed in a7c233a.